File: airbyte-config-oss/config-models-oss/src/main/java/io/airbyte/configoss/AirbyteConfigValidator.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.configoss;

import io.airbyte.validation.json.AbstractSchemaValidator;
import java.nio.file.Path;

public class AirbyteConfigValidator extends AbstractSchemaValidator<ConfigSchema> {

  final public static AirbyteConfigValidator AIRBYTE_CONFIG_VALIDATOR = new AirbyteConfigValidator();

  @Override
  public Path getSchemaPath(final ConfigSchema configType) {
    return configType.getConfigSchemaFile().toPath();
  }

}

File: airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/configoss/AirbyteConfigValidator.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.configoss

import io.airbyte.validation.json.AbstractSchemaValidator
import java.nio.file.Path

class AirbyteConfigValidator : AbstractSchemaValidator<ConfigSchema>() {
    override fun getSchemaPath(configType: ConfigSchema): Path {
        return configType.configSchemaFile.toPath()
    }

    companion object {
        val AIRBYTE_CONFIG_VALIDATOR: AirbyteConfigValidator = AirbyteConfigValidator()
    }
}
